### ISSUE TITLE
Add non-root var (UID/GID) to run docker-nextcloud

### ISSUE THREAD

**Issue by SckyzO** [crazy-max/docker-nextcloud#6]
Hi, 

Thx for your Nextcloud docker version, 
I use it and i love it :) 

I would like to run containers without root access 

I would like to add Environment variables
- UID : nextcloud user id
- GID : nextcloud group id

Do you think it is possible ? 

Regards,

**Comment by crazy-max** [comment]
Hi @SckyzO, yes we can definitely improve this. Thanks for your support!

**Comment by enchained** [comment]
Should running docker-compose up -d via sudo be enough as "running nextcloud container as root" criteria? 
I have other services like transmission container writing to host system folder owned by root. So I just used root ids as env vars for such containers. I mounted that host system folder as volume for my nextcloud container in compose, and then enabled external storage app and mounted it as local in the external storage settings. I can see all the files from that external storage just fine, but can't upload or change anything, there's a warning "You don’t have permission to upload or create files here". Any idea on what can I do to fix that?

**Comment by enchained** [comment]
To fix that, instead of trying to run as root, I had to see ids of user, who owns main cloud files inside the container:
`docker exec -it -u root compose_app_1 ls -la /data/data/nextcloud-admin/files`
Then on the host system I created a user, then did `usermod`/`groupmod` to make its ids the same. After that I just granted permissions to that user via `setfacl` on my external storage folder on the host system. After that I was able to upload to my external storage in nextcloud.
